ambry-cloud/src/main/java/com/github/ambry/cloud/azure/ConnectionStringBasedStorageClient.java

```diff
@@ -90,6 +95,16 @@ protected BlobServiceClient buildBlobServiceSyncClient(HttpClient httpClient, Co
         .buildClient();
   }
 
+  @Override
+  protected TableServiceClient buildTableServiceClient(HttpClient httpClient, Configuration configuration,
+      RetryOptions retryOptions, AzureCloudConfig azureCloudConfig) {
+    return new TableServiceClientBuilder().connectionString(azureCloudConfig.azureTableConnectionString)
```

Comment:
By default, azureTableConnectionString is null. 
Looks like getTableServiceClient will throw exception to the constructor.
Do you test the case when  azureTableConnectionString is null?

Reply:
Added a method `validateTableServiceConfigs` which will fail VCR startup if the table-conn-str is null and we are using a conn-str client.  The conn-str varies depending on the azure account, so we cannot hardcode it here.